generator/testdata/google/example/library/v1/library_service.gapic.h.baseline

```diff
@@ -66,24 +66,15 @@ class LibraryService final {
   std::unique_ptr<google::gax::RetryPolicy> retry_policy_;
   std::unique_ptr<google::gax::BackoffPolicy> backoff_policy_;
 
+
   // Note: conservatively assume no methods are idempotent.
   //       This will eventually be set from annotations.
-  static constexpr google::gax::MethodInfo CreateBook_info_ = {"CreateBook", google::gax::MethodInfo::RpcType::NORMAL_RPC};
-  // Note: conservatively assume no methods are idempotent.
-  //       This will eventually be set from annotations.
-  static constexpr google::gax::MethodInfo GetBook_info_ = {"GetBook", google::gax::MethodInfo::RpcType::NORMAL_RPC};
-  // Note: conservatively assume no methods are idempotent.
-  //       This will eventually be set from annotations.
-  static constexpr google::gax::MethodInfo ListBooks_info_ = {"ListBooks", google::gax::MethodInfo::RpcType::NORMAL_RPC};
-  // Note: conservatively assume no methods are idempotent.
-  //       This will eventually be set from annotations.
-  static constexpr google::gax::MethodInfo DeleteBook_info_ = {"DeleteBook", google::gax::MethodInfo::RpcType::NORMAL_RPC};
-  // Note: conservatively assume no methods are idempotent.
-  //       This will eventually be set from annotations.
-  static constexpr google::gax::MethodInfo UpdateBook_info_ = {"UpdateBook", google::gax::MethodInfo::RpcType::NORMAL_RPC};
-  // Note: conservatively assume no methods are idempotent.
-  //       This will eventually be set from annotations.
-  static constexpr google::gax::MethodInfo GetBigBook_info_ = {"GetBigBook", google::gax::MethodInfo::RpcType::NORMAL_RPC};
+  static constexpr google::gax::MethodInfo create_book_info = {"CreateBook", google::gax::MethodInfo::RpcType::NORMAL_RPC};
```

Comment:
Consider this alternative:
```C++
static google::gax::MethodInfo const& create_book_info() {
  static google::gax::MethodInfo info = {....};
  return info;
}
```

I am not sure it is "better", but at least does not require a separate declaration and definition.

Reply:
I like the additional guarantees provided by making the structs `constexpr`.
What about returning by value,
```
static constexpr google::gax::MethodInfo create_book_info() {
  return google::gax::MethodInfo{...};
}
```
?

In an ideal world, there would be better communication between the compiler and linker, and this double description wouldn't be necessary.

Reply:
Sticking with the static attributes since it clarifies intent.